### DISCUSSION TITLE
Trigger Execution of Python Script on main PC using Micropython on microcontroller

### DISCUSSION THREAD

**Discussion by Justin14He** [micropython/micropython#11858] [Using MicroPython]
Hello everyone,

I am interested in using an interrupt detected on the Raspberry Pi Pico to send a signal to my personal computer through the USB, triggering a python script to run from the computer. I've probed the internet looking for a way to do this, but have been unsuccessful. Do any of you have a tip or guidance on this matter?

Thank you for your support in advance! And apologies if my post etiquette is imperfect - I am new to the community.

**Comment by bixb922** [comment]
When you print() in Micropython, that info gets sent to the USB as a COM Port. You can open and read that COM port on the PC side with Python. Mpremote will show the name of the Com port on the PC.
There are interesting libraries like Belay that do this on a much more integrated way, see https://github.com/BrianPugh/belay
